gossipd/gossip.c

```diff
@@ -180,6 +347,41 @@ static bool send_peer_with_fds(struct peer *peer, const u8 *msg);
 static void wake_pkt_out(struct peer *peer);
 static bool try_reach_peer(struct daemon *daemon, const struct pubkey *id);
 
+
+
+
+// called when we want to connect to TOR SOCKS5
+static struct io_plan *io_tor_connect(struct io_conn *conn, struct reaching *reach)
+{
+	static struct addrinfo *ai_tor;
+	static char *port_addr;
+
+	reach_tor.port=htons(reach->addr.port);
+
+	port_addr = tal_fmt(NULL,"%u",reach->daemon->tor_proxy_port);
```

Comment:
Why is the context here NULL? Should it not be either `conn` or `reach` or a `const tal_t *tmpctx = tal_tmpctx(NULL);` that you `tal_free(tmpctx);` before returning?